src/Text/Pandoc/Readers/HTML.hs

```diff
@@ -342,7 +342,20 @@ pListItem = do
                                 [Span (ident, [], []) ils] : xs)
                            _ -> B.divWith (ident, [], []) bs
   maybe id addId (lookup "id" attr) <$>
-    pInTags "li" block
+    pInTags "li" (pLabel' <|> block)
+  where 
+    pLabel' :: PandocMonad m => TagParser m Blocks
+    pLabel' = try $ pInTag TagsRequired "label" pInput'
+    pInput' :: PandocMonad m => TagParser m Blocks
+    pInput' = do
+          TagOpen _ attr' <- lookAhead $ pSatisfy $ matchTagOpen "input" [("type","checkbox")]
+          let attr = toStringAttr attr'
+          let isChecked = isJust $ lookup "checked" attr
+          let escapeSequence = B.Str $ if isChecked then "\9746" else "\9744"
+          _ <- pSelfClosing (== "input") (const True)
+          bs <- inline
+          return $ B.Many . Seq.singleton $ B.Plain ([escapeSequence, B.Space ] <> B.toList bs)
```

Comment:
Here you can just use the function `plain` from Text.Pandoc.Builder to avoid these conversions:
```hs
return $ B.plain (escapeSequence <> B.space) <> bs
```
You'll also need to use B.str for escapeSequence:
```hs
let escapeSequence = B.str $ if isChecked ...etc.
```
Basically `B.str` = `B.Many . Seq.singleton . B.Str`.

Reply:
Sure! Just cleaned the code up and it looks a lot better!